src/components/sbb-checkbox/sbb-checkbox.tsx

```diff
@@ -0,0 +1,138 @@
+import { Component, Prop, h, Event, EventEmitter, Watch, JSX } from '@stencil/core';
+import { InterfaceCheckboxAttributes } from './sbb-checkbox.custom';
+
+let nextId = 0;
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-checkbox.scss',
+  tag: 'sbb-checkbox',
+})
+export class SbbCheckbox {
+  private _checkbox: HTMLInputElement;
+
+  /** Whether the checkbox is checked. */
+  @Prop({ mutable: true }) public checked: boolean;
+
+  /** Value of checkbox. */
+  @Prop() public value?: string;
+
+  /** Name of the checkbox */
+  @Prop() public name?: string;
+
+  /** Id of the internal input element - default id will be set automatically. */
+  @Prop() public inputId = `sbb-checkbox-${++nextId}`;
+
+  //** the svg for the true state */
+  @Prop() public labelIcon: '';
+
+  /** The disabled prop for the disabled state. */
+  @Prop() public disabled!: boolean;
+
+  /** The required prop for the required state. */
+  @Prop() public required?: boolean;
+
+  /** Whether the checkbox is tristated. */
+  @Prop() public tristated?: boolean;
+
+  /** The label position relative to the labelIcon. Defaults to false */
+  @Prop() public labelPosition: InterfaceCheckboxAttributes['labelPosition'] = 'after';
+
+  /** Whether the checkbox label has spacing to the labelIcon. */
+  @Prop() public labelSpace = false;
+
+  /** The aria-label prop for the hidden input. */
+  @Prop() public acceccibilityLabel?: string;
+
+  /** The aria-labelledby prop for the hidden input. */
+  @Prop() public acceccibilityLabelledby?: string;
+
+  /** The aria-describedby prop for the hidden input. */
+  @Prop() public acceccibilityDescribedBy?: string;
+
+  /** Event for emiting whenever selection is changed. */
+  @Event() public sbbChange: EventEmitter;
+
+  /** render the svg acording to the state */
+  private _renderStateIcon(): JSX.Element {
+    if (this.tristated) {
+      return (
+        <svg xmlns="http://www.w3.org/2000/svg" width="8" height="2" viewBox="0 0 8 2" fill="none">
+          <path
+            d="M1 1H7"
+            stroke="#EB0000"
```

Comment:
Please apply color from CSS variables. There are going to be security critical application use cases, which require being able to change red colors to a different color.